src/python.cpp

```diff
@@ -1251,110 +1251,140 @@ SEXP py_to_r(PyObject* x, bool convert) {
 
 }
 
+/* stretchy list, modified from R sources
+   CAR of the list points to the last cons-cell
+   CDR points to the first.
+*/
+
+SEXP NewList(void) {
+  SEXP s = Rf_cons(R_NilValue, R_NilValue);
+  SETCAR(s, s);
+  return s;
+}
+
+/* Add named element to the end of a stretchy list */
+void GrowList(SEXP args, SEXP tag, SEXP dflt) {
```

Comment:
Do we need to worry about how these arguments are protected here?

Reply:
Good catch! I added a `PROTECT(dflt)` before the `Rf_cons()` call.